### ISSUE TITLE
Search filter issues

### ISSUE THREAD

**Issue by thatbudakguy** [Princeton-CDH/derrida-django#203]
- [x] Make filter text & button clickable & use `cursor: pointer;`, maybe set an aria role to help
- [x] Add hover states where possible, using zeplin as a reference
- [x] Escape key to close filter and submit the form
- [x] Wrap checkboxes in a `<fieldset>` if possible
- [x] Ensure everything is tab-navigable

@bwhicks is our star tester for this one ⭐️

**Comment by thatbudakguy** [comment]
https://www.w3.org/TR/2017/WD-wai-aria-practices-1.1-20170628/examples/listbox/listbox.html

**Comment by thatbudakguy** [comment]
there's a variety of things remaining to address here, some of which are a11y-related and some not.

- `cursor: pointer;` should probably be applied when the user is hovering over any part of the field, including the blank line below it, which doesn't have it currently.
- likewise, hover state (making the blank line blue) should be applied when the user is hovering over any part of the field - currently it only appears when the user hovers over the line.
- when the user clicks the blank line below a field, the field will open up and then immediately close again. after the first click, it will behave normally. this only happens if the user clicks the blank line area rather than the name of the field or the dropdown arrow.

a11y stuff:

- <kbd>esc</kbd> doesn't close the field or submit the form.
- when the user tabs between fields, all previously focused fields will stay open. this clutters the display and makes it hard to read/see what options are actually active. we probably need to add a `blur()` handler for the fields to close them once focus leaves them.
- when the user focuses a field, the next item in the tab order will be the content of that field rather than the next field. this creates a keyboard trap inside fields with a lot of content, like the author field (which is first, on the library page) - the user is forced to tab through every single author checkbox in order to get to the next field.

sadly, the last two issues mentioned are fairly bad for a11y and also likely not trivial to address because we probably have to implement focus/tabindex management in a way that interacts with the material design js/css.

tested in chrome 79 and ff 73.
